src/callbacks.c

```diff
@@ -1,4 +1,12 @@
 #include "curl-common.h"
+#include "utils.h"
+
+int default_callback_progress(void *dummy,
+                              double dltotal, double dlnow,
+                              double ultotal, double ulnow)
+{
+  return !(R_ToplevelExec(check_interrupt_fn, NULL));
```

Comment:
Yes, but again only manually and a few times, I did not see any problems. Do you know of a good way to automate tests involving signals? I guess on Unix-like systems one could write a script that issues signals to an R process on some interval...